applications/structure/bar/application.h

```diff
@@ -319,6 +319,27 @@ class Application : public ApplicationBase<dim, Number>
     this->param.preconditioner      = preconditioner;
     this->param.multigrid_data.type = MultigridType::phMG;
 
+    this->param.multigrid_data.coarse_problem.amg_data.amg_operator_type =
+      AMGOperatorType::Elasticity;
+
+#ifdef DEAL_II_WITH_TRILINOS
+    this->param.multigrid_data.coarse_problem.amg_data.ml_data.elliptic = true;
+    bool const mg_involves_p = (this->param.multigrid_data.type == MultigridType::pMG or
+                                this->param.multigrid_data.type == MultigridType::hpMG or
+                                this->param.multigrid_data.type == MultigridType::phMG);
+    this->param.multigrid_data.coarse_problem.amg_data.ml_data.higher_order_elements =
+      (preconditioner == Preconditioner::AMG or
+       (preconditioner == Preconditioner::Multigrid and not mg_involves_p)) ?
+        this->param.degree > 1 :
+        false;
```

Comment:
I suggest to remove this parameter. I did experiment with it in 2008, and with the state of ML at that time it was useful. However, improvements in ML around that time made the parameter obsolete, and the parameter just sets the parameter "aggregation: type" to "Uncoupled" which is at that value anyway: https://github.com/dealii/dealii/blob/14276d45ac74825eb1a3116f73bda03680855465/source/lac/trilinos_precondition_ml.cc#L75-L84

Reply:
I just removed it for now. `aggregation: type` is always uncoupled by default anyways? For `elliptic` and `not higher_order`, we set it to decoupled, but I read the comment in the deal.II part you sent, as if it was the default anyways? And that part just remains as kind of a reminder? I just set the parameter to not make anything wrong, you know. I have no clue about the ML intrinsics/settings etc. I just know, that it had barely any or even no effect setting this parameter.

Reply:
Yes, this is what I read from https://ftp.mcs.anl.gov/pub/pdetools/nightlylogs/xsdk/xsdk-configuration-tester/packages/trilinos/packages/ml/doc/mlguide.pdf (section 6.4.5, page 23) as being the default anyway. It is there as a reminder indeed. This is why I suggest removing it.